disk_objectstore/container.py

```diff
@@ -1893,10 +1848,15 @@ def _vacuum(self) -> None:
 
         (See also description in issue #94).
         """
-        engine = self._get_cached_session().get_bind()
-        engine.execute("VACUUM")
+        # VACUUM cannot be performed from within a transaction
+        # see: https://github.com/sqlalchemy/sqlalchemy/discussions/6959
+        session = self._get_cached_session()
+        session.execute(text("COMMIT"))
+        session.execute(text("VACUUM"))
+        # ensure sqlalchemy knows to open a new transaction for the next execution
+        session.commit()
```

Comment:
What is the difference between `session.execute(text("COMMIT"))` and `session.commit()` and why is it necessary to make it differently here?

Reply:
The difference is that this direct execution bypasses sqlalchemy’s auto-transaction mechanics, and stops it from starting a new transaction before executing the vacuum (which cannot be executed within a transaction)